jade-repl.el

```diff
@@ -194,31 +194,98 @@ When ERROR is non-nil, use the error face."
     (jade-repl-insert-prompt)
     (run-hooks 'jade-repl-evaluate-hook)))
 
-(defun jade-repl-emit-console-message (string &optional level)
-  "Emit a console message STRING.
-LEVEL is a string representing the logging level, it can be
-\"log\", \"warn\", \"debug\" or \"error\"."
+(defun jade-repl-emit-console-message (message)
+  "Emit a console message.
+MESSAGE is a map (alist/hash-table) with the following keys:
+  level		severity level (can be log, warning, error, debug)
+  text		message text to be displayed
+  type		type of message
+  url		url of the message origin
+  line		line number in the resource that generated this message
+  parameters	message parameters in case of the formatted message
+
+MESSAGE must contain `text' or `parameters.'. Other fields are
+optional."
   (with-current-buffer (jade-repl-get-buffer)
     (save-excursion
-      (let* ((error (string= level "error"))
-             (face (when error 'jade-repl-error-face))
-             (message (if level
-                          (concat level ": " string)
-                        string)))
+      (let ((level (or (map-elt message 'level) ""))
+            (text (map-elt message 'text))
+            (parameters (map-elt message 'parameters))
+            (url (map-elt message 'url))
+            (line (map-elt message 'line)))
         (goto-char jade-repl-output-end-marker)
-        (insert "\n")
         (set-marker jade-repl-output-start-marker (point))
-        (insert
-         (ansi-color-apply
-          (propertize message
-                      'font-lock-face (or face 'jade-repl-stdout-face)
-                      'rear-nonsticky '(font-lock-face))))
+        (jade-repl--emit-values text
+                                parameters
+                                level url line)
```

Comment:
Sure, there was a map over parameters which I removed and forgot to align the args.